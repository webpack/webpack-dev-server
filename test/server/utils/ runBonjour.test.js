'use strict';

const runBonjour = require('../../../lib/utils/runBonjour');

describe('runBonjour', () => {
  let mock;
  let publish = jest.fn();
  let unpublishAll = jest.fn();

  beforeAll(() => {
    mock = jest.mock('bonjour', () => () => {
      return {
        publish,
        unpublishAll,
      };
    });
  });

  afterEach(() => {
    publish = jest.fn();
    unpublishAll = jest.fn();
  });

  afterAll(() => {
    mock.mockRestore();
  });

  it('should call bonjour.publish', () => {
    runBonjour({
      port: 1111,
    });

    expect(publish.mock.calls[0][0]).toMatchSnapshot();
  });
});
