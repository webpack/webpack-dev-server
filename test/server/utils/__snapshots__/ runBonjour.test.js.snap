// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runBonjour should call bonjour.publish 1`] = `
Object {
  "name": "Webpack Dev Server",
  "port": 1111,
  "subtypes": Array [
    "webpack",
  ],
  "type": "http",
}
`;
